eZ/Publish/API/Repository/Tests/SearchService/DeleteTranslationTest.php

```diff
@@ -123,4 +126,72 @@ public function testDeleteContentTranslation(): void
         $searchResult = $this->findContent('Kontakt', 'ger-DE');
         $this->assertEquals(0, $searchResult->totalCount);
     }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testDeleteContentTranslationWithContentRemovePolicy(): void
+    {
+        $repository = $this->getRepository();
+        $testContent = $this->createTestContentWithLanguages(
+            [
+                'eng-GB' => 'Contact',
+                'ger-DE' => 'Kontakt',
+                'eng-US' => 'Contact',
+            ]
+        );
+
+        $repository->getPermissionResolver()->setCurrentUserReference($this->provideUserWithContentRemovePolicies());
+
+        $contentService = $repository->getContentService();
+        $searchResult = $this->findContent('Contact', 'eng-US');
+        $this->assertEquals(1, $searchResult->totalCount);
+
+        $contentService->deleteTranslation($testContent->contentInfo, 'eng-US');
+        $this->refreshSearch($repository);
+        $searchResult = $this->findContent('Contact', 'eng-US');
+        $this->assertEquals(0, $searchResult->totalCount);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testPreventTranslationDeletionIfNoAccess(): void
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $testContent = $this->createTestContentWithLanguages(
+            [
+                'eng-GB' => 'Contact',
+                'ger-DE' => 'Kontakt',
+                'eng-US' => 'Contact',
+            ]
+        );
+
+        $repository->getPermissionResolver()->setCurrentUserReference($this->provideUserWithContentRemovePolicies());
```

Comment:
CS: this line is too long.

Reply:
Resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3036/commits/9285afbc3645b5837f2c88f811fb8dc019b1e3ba